### ISSUE TITLE
Add ability to work on lists/maps of resources

### ISSUE THREAD

**Issue by moritzheiber** [aelsabbahy/goss#183]
I would love to be able to loop over a range of items/resources that are quite similar in nature, e.g. a list of packages or a list of users.

I would propose something along the lines of

```yaml
user_map:
  user_a: "user name"
  user_b: "user name"

package_list:
- package_a
- package_b
```

```
# Map example
{{- range $u, $name := user_map -}}
user:
  $u:
    exists: true
    groups:
    - $u
    home: /home/$u
    shell: /sbin/bash
{{- end -}}

# List example
{{- range $p := package_list -}}
package:
  $p:
    installed: true
{{- end -}}
```

I think it would be quite convenient.

**Comment by aelsabbahy** [comment]
Would the solution proposed https://github.com/aelsabbahy/goss/issues/136#issuecomment-242773611 resolve this issue?

**Comment by moritzheiber** [comment]
It has the same spirit behind it at least 😊

I don't see any room for iterative processing in his approach though, or are you planning on implementing a whole range of templating functions?

**Comment by aelsabbahy** [comment]
Thinking of a simple implementation that basically allows the goss yaml/json file to be a golang [text/template](https://golang.org/pkg/text/template/) and allowing the user to provide an arbitrary json/yaml vars definition file.

**Comment by pysysops** [comment]
I wonder if iteration through resources might be easily dealt with by allowing the use of comma separated lists as resource titles?

Example:
```yaml
package:
  "package_a, package_b":
    installed: true
```

Perhaps the use of a special "title" (similar to Puppet) could be implemented:
```yaml
user:
  "adam, john, tom, pete":
    exists: true
    groups:
    - $title
    home: /home/$title
    shell: /sbin/bash
```

There is some value in the template package but thought I'd offer a different approach that could work.

**Comment by aelsabbahy** [comment]
Released in [v0.3.0](https://github.com/aelsabbahy/goss/releases/tag/v0.3.0)
